01-Login/Auth0Sample/HomeViewController.m

```diff
@@ -32,18 +32,32 @@ @interface HomeViewController ()
 
 @implementation HomeViewController
 
+static BOOL logged = false;
+
 - (IBAction)showLoginController:(id)sender {
     HybridAuth *auth = [[HybridAuth alloc] init];
 
-    [auth showLoginWithScope:@"openid" connection:nil callback:^(NSError * _Nullable error, A0Credentials * _Nullable credentials) {
-        dispatch_async(dispatch_get_main_queue(), ^{
-            if (error) {
-                NSLog(@"Error: %@", error);
-            } else if (credentials) {
-                [self showAlertWithMessage:[NSString stringWithFormat:@"Success, Access Token: %@", [credentials accessToken]]];
-            }
-        });
-    }];
+    if (!logged) {
+        [auth showLoginWithScope:@"openid"
+                      connection:nil callback:^(NSError * _Nullable error,
+                                                                     A0Credentials * _Nullable credentials) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                if (error) {
+                    NSLog(@"Error: %@", error);
+                } else if (credentials) {
+                    [self showAlertWithMessage:[NSString stringWithFormat:@"Success, Access Token: %@", [credentials accessToken]]];
+                    [sender setTitle:@"Log out" forState:UIControlStateNormal];
+                    logged = true;
+                }
+            });
+        }];
+        
+    }else{
+        [auth userLogout];
```

Comment:
The async method called in this function has a callback to let you know the the logout endpoint was called OKAY. You should probably use that outcome to decide on subsequent logic.